src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.test.js

```diff
@@ -1,151 +1,192 @@
-import convertInclude from '.';
+import convertInclude, { encodeHTML } from '.';
+
+export const vjMarkup = `<div>Visual Jounalism Markup</div><script type="text/javascript" src="localhost/vj.js"></script>`;
+
+export const idt2Markup = `<div>IDT 2 Markup</div><script type="text/javascript" src="localhost/idt2.js"></script>`;
+
+export const idt1Markup = `<div>IDT 1 Markup</div><script type="text/javascript" src="localhost/idt1.js"></script>`;
 
 describe('convertInclude', () => {
-  it('should convert an include block to an idt1 block', () => {
+  afterEach(() => {
+    fetch.resetMocks();
+  });
+
+  it('should fetch and convert an include block to an idt1 block', async () => {
+    fetch.mockResponse(() => Promise.resolve(idt1Markup));
     const input = {
       required: false,
       tile: 'A quiz!',
-      href:
-        '/indepthtoolkit/quizzes/sistema_solar_Quiz_Mundo/syndicated?app-clickable=true&app-image=http%3A%2F%2Fnews.files.bbci.co.uk%2Fvj%2Flive%2Fidt-images%2Fquizzes-quiz_cta%2Fstart_geh2q.png',
+      href: '/indepthtoolkit/quizzes/123-456',
       platform: 'highweb',
       type: 'include',
-      url: 'https://bbc.com/idt1.html',
     };
     const expected = {
       type: 'include',
       model: {
-        href: 'https://bbc.com/idt1.html',
+        href: '/indepthtoolkit/quizzes/123-456',
         required: false,
         tile: 'A quiz!',
         platform: 'highweb',
         type: 'idt1',
+        html: encodeHTML(idt1Markup),
```

Comment:
We are using the same code to generate the expected markup as is used in `convertInclude` itself. Should we hardcode the encoded markup in a _set of new consts_ at the top of this file? That way if we change the encoding logic later we would catch any regressions e.g. we accidently didn't encode

Reply:
yeah, makes sense